### ISSUE TITLE
Describe installing CaaSP from SCC

### ISSUE THREAD

**Issue by r0ckarong** [SUSE/doc-caasp#205]
We have the SCC setup now, meaning CaaSP is installable now as an extension.

The product is in the Alpha stage, meaning only internal SUSE employees can access to it. So we need to document 2 different things:

1- How to use SCC as an employee for the Alpha (release notes or announcement email)
2- How to use SCC in general (deployment)

Later on, we will have to adapt 1 to Beta users.

# How to use SCC as an employee for the Alpha

1- Go to https://scc.suse.com/admin/products/1863 and click on "I want a reg code!"
2- Follow the steps. Click on "Activate ..." Warning: Do not select Novell Inc as an organization for performance usage.
3- Now you should have a subscription code in https://scc.suse.com/admin/products/1863

Use this code in "How to use SCC in general"

# How to use SCC in general

You need a registration code. With that, you can use "SUSEConnect -p caasp/4.0/x86_64 -r REGCODE " and then install the packages .... (more info on what to install?)

# How to use the beta codes

We will have a list of beta codes. PM should share them with the customers? How a customer requests a code? in the mailing list???

**Comment by r0ckarong** [comment]
@jordimassaguerpla Currently weird situation because YaST defaults to btrfs for partitions but the images are supposed to use XFS. I am trying to find out what the right behavior is and what to do about this.

**Comment by flavio** [comment]
We don't actually care that much about the filesystem that is being used on the system.

To provide the bigger picture:

  * We are not using transactional-update anymore, so we don't need btrfs
  * the container engine (docker, crio, podman, all of them) are nowadays working great with overlayfs as graph driver
  * overlayfs can work on top of any kind of underlying filesystem. Translated: `/var/lib` can be using btrfs, xfs, ext4 and nothing would matter to the container engine
  * overlayfs is our preferred choice as a graph driver for container engines because it has good performances and it reduces the codebase we have to maintain
  * there are **rumors** about performance issues with overlayfs on top of btrfs. @cyphar is aware of this issue and is going to look into that
  * in the meantime the container engine (podman, docker, crio,...) automatically picks the btrfs graph driver if `/var/lib` is on a btrfs volume, otherwise it will use overlayfs

Long story short: we don't have to request a special partitioning setup or filesystems for v4 nodes.

Provocative question (but serious one): do we actually need to explain our customers how to install a SLES system? I think they can use the existing documentation of SLES to get there, especially given we don't need any special setup.

I hope that helps.

**Comment by r0ckarong** [comment]
@flavio 
> Provocative question (but serious one): do we actually need to explain our customers how to install a SLES system? I think they can use the existing documentation of SLES to get there, especially given we don't need any special setup. 

Actually that is a good point. In this case it's special because we explain how to create a template for VMWare for the deployment and that needs SLES.

Who can make the call that for other scenarios that just rely generally on SLES we will not describe their steps anymore? Klaus?

**Comment by flavio** [comment]
> Actually that is a good point. In this case it's special because we explain how to create a template for VMWare for the deployment and that needs SLES.

This is needed now, but it won't be on the medium term. That's because our customers are going to pick the official pre-built SLES images for VMware and just use them. They cannot do that right now because a package is missing (cloud-init), but the package is going to be part of the next build of the image.

All of that to say: don't put too much efforts into writing this documentation, we will hopefully drop all of that soon.

> Who can make the call that for other scenarios that just rely generally on SLES we will not describe their steps anymore? Klaus?
>
> I'm happy to drop anything SLES related and rely on the SLES upstream docs, makes our life simpler in many ways.

I think @kkaempf, as product owner, is the best person to answer this question.

**Comment by Martin-Weiss** [comment]
Please keep in mind that the "VMware templating functionality" for base-os does not fit for the majority of deployments so far. Background: when ever we add or replace a node in the cluster the new / replaced node needs to have the same patchlevel than the rest of the cluster. In case the templates are not updated constantly together with applying updates to the cluster - the template needs to be cloned and then patched - and this process often takes more time than installing the server automatically via AutoYaST. It is also highly recommended to use AutoYaST for creating a template image to ensure it is 100% reproducible - so in most cases AutoYaST is available anyway. AutoYaST also gives much more flexibility for future adjustments so that the majority of the customers I have been dealing with were deciding for using AutoYaST instead of templating/cloning servers for their cluster deployments.
With or without SUSE Manager - the bare metal provisioning using AutoYaST can be 100% automated including usage of VSphere APIs using Cobbler, PXE or customized ISOs.

**Comment by flavio** [comment]
Thanks for the feedback, don't worry this is a temporary solution until the cloud-init package is made part of the official SLES image for VMware.

However keep in mind that installing CaaSP v4 with AutoYaST is still going to be supported. Even if a customer will use this deployment mechanism on something that is not bare metal.

**Comment by kkaempf** [comment]
> Who can make the call that for other scenarios that just rely generally on SLES we will not describe their steps anymore? Klaus?

Yes, I can do this. We do NOT describe how to install SLES. A pointer to the respective SLES documentation is sufficient.

And when you use @kkaempf in an issue comment next time, it gets my attention immediately :wink:
